### PR TITLE
fix(parse-i3s-tile-content): I3S conversion issue without draco

### PR DESCRIPTION
**Pull request by HeimEndyd**
An error occurred when a scene with materials without textures was parsed. When trying to get a uv0 buffer, we went beyond the buffer boundary. Parsing 'Geometry buffer' depends on 'ordering'. In one scene some materials may have texture while others may not. Because of this, we cannot guarantee to remove the error by affecting only the 'ordering', removing the uv0 from it. Therefore, we will simply skip the step with getting uv0 if getting it goes beyond the boundaries of the buffer.

spec: https://github.com/Esri/i3s-spec/blob/master/docs/1.8/defaultGeometrySchema.cmn.md

### CODE REVIEW

**Comment by belom88** [comment]
You still need to remove `uv0` from `defaultGeometrySchema` if there is no any texture in the whole layer. ArcGis won't mind this logic.

**Review by belom88 — CHANGES_REQUESTED** [review]
You still need to remove uv0 from defaultGeometrySchema if there is no any texture in the whole layer. ArcGis won't mind this logic.

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 1]
```diff
@@ -556,6 +566,10 @@ export default class I3SConverter {
     };
 
     for (const resources of resourcesData || [emptyResources]) {
+      if (!this.layersHasTexture && resources.texture) {
+        this.layersHasTexture = true;
+      }
```
Comment (belom88) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:571]:
```suggestion
      this.layersHasTexture = this.layersHasTexture || resources.texture;
```

**Review by belom88 — APPROVED** [review]
One nit comment

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts [hunk 2]
```diff
@@ -317,44 +317,46 @@ function normalizeAttributes(
...
+          case 'position':
+          case 'region':
+          case 'normal':
+          default:
+        }
 
-      byteOffset = byteOffset + count * valuesPerElement * sizeOf(valueType);
+        byteOffset = byteOffset + attributeCount * valuesPerElement * sizeOf(valueType);
+      } else if (attribute !== 'uv0') {
+        break;
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts:358]: Nit: We generally avoid `break` statements. It is OK if you can't find another way.

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 3]
```diff
@@ -226,6 +228,14 @@ export default class I3SConverter {
     await this._convertNodesTree(root0, sourceRootTile, parentId, boundingVolumes);
 
     this.layers0!.materialDefinitions = this.materialDefinitions;
+
+    if (!this.layersHasTexture) {
+      this.layers0!.store.defaultGeometrySchema.ordering =
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:233]: No exclamation marks please

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 4]
```diff
@@ -556,6 +566,8 @@ export default class I3SConverter {
     };
 
     for (const resources of resourcesData || [emptyResources]) {
+      this.layersHasTexture = this.layersHasTexture || !!resources.texture;
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:569]:
```suggestion
      this.layersHasTexture = this.layersHasTexture || Boolean(resources.texture);
```

**Review by ibgreen — APPROVED** [review]

**Pull request merged**